src/libjasper/jpc/jpc_t2dec.c

```diff
@@ -195,10 +195,21 @@ static int jpc_dec_decodepkt(jpc_dec_t *dec, jas_stream_t *pkthdrstream, jas_str
 			}
 			if (jpc_ms_gettype(ms) != JPC_MS_SOP) {
 				jpc_ms_destroy(ms);
-				jas_eprintf("missing SOP marker segment\n");
+				jas_eprintf("cannot get (SOP) marker segment\n");
+				return -1;
+			}
+			/* checking the packet sequence number */
+			if (dec->numpkts % (UINT_FAST16_MAX + 1) != ms->parms.sop.seqno) {
+				jas_eprintf("incorrect packet sequence number %d was found, but expected %d\n",
+					ms->parms.sop.seqno, dec->numpkts % (UINT_FAST16_MAX + 1));
+				jpc_ms_destroy(ms);
 				return -1;
 			}
 			jpc_ms_destroy(ms);
+		} else {
+			/* SOP marker segment may not be present 
+			   even if the second LSB of the SPcod is set to "1" */
+			jas_eprintf("warning: missing SOP marker segment\n");
```

Comment:
Is this worth logging?
I hate all `jas_eprintf()` calls because this prints to the application's stderr, which may or may not be usable, but most likely will never be read by the user, and cannot be evaluated programmatically by the application.
Aside from this general misdesign in JasPer's API, is it worth logging this event?

Reply:
If a codestream is accidentally broken, some application would implement "resilience mode" with SOP and EPH marker segments. However, I don't know there is such an application. I think we can simply delete this logging without any severe problem.